dts/bindings/dsa/microchip,ksz8794.yaml

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) 2020 DENX Software Engineering GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    KSZ8794 ethernet switch with SPI interface
+
+compatible: "microchip,ksz8794"
+
+include: [spi-device.yaml]
+properties:
+    dsa-master-port:
+       type: phandle
+       required: false
+       description: Phandle to master port.
+    dsa-slave-ports:
+       type: int
+       required: false
+       description: Number of slave ports on the switch
+    spi-cpha:
+       type: boolean
+       required: false
+       description: |
+          Set to indicate phase starts with asserted half-phase (CPHA=1).
+          For this driver using this property requires also using cpol.
+    spi-cpol:
+        type: boolean
+        required: false
+        description: |
+          Set to indicate clock leading edge is falling (CPOL=1).
+          For this driver using this property requires also using cpha.
+    reset-gpios:
+        type: phandle-array
+        required: false
+        description: |
+          The pin is asserted for 10ms during boot to reset the KSZ8794.
+    workaround:
+        type: int
+        required: false
+        description: |
+          Define the applied workaround for the switch used for
+          short connections. Use bitmask to select the workaround or more
+          0x01: Short Cable Problems with the KSZ8795 Family
+          0x02: 1) CAT-5E/6 Short Cable with a Link Issue for the KSZ8795 Family
+          0x04: 2) CAT-5E/6 Short Cable with a Link Issue for the KSZ8795 Family
+        enum:
+          - 0x01
+          - 0x02
+          - 0x04
```

Comment:
Hmm, I wasn't asking for this to be an `enum`. The documentation says it's a bitmask, which makes it look like 0x7 is a valid value. Using an `enum` would cause an error if the user tries that though. Are you sure?

Reply:
Ok, I've re-check it. In this particular case - the "workaround" property shouldn't be an enum, as it is valid to have 0x7 as the value. This enum shall be removed.
However, IMHO it looks correct to have enum in the 'mii-lowspeed-drivestrength' property, as set current values are distinct.